Analyser/AmplFilter/src/com/Main.java

```diff
@@ -0,0 +1,87 @@
+package com.roboy;
+
+import java.io.*;
+import java.math.BigInteger;
+
+public class Main
+{
+    public static void main(String[] args)
```

Comment:
also, add comment explaining what is going on, at least general ones for each function